yt_dlp/extractor/huya.py

```diff
@@ -139,73 +141,106 @@ def encrypt(self, params, stream_info, stream_name):
-    },
-        {
+    }, {
         'url': 'https://www.huya.com/video/play/556054543.html',
         'info_dict': {
             'id': '556054543',
             'ext': 'mp4',
             'title': '我不挑事 也不怕事',
-            'thumbnail': r're:https?://.*\.jpg',
-            'duration': 1864,
+            'categories': ['英雄联盟'],
+            'description': 'md5:58184869687d18ce62dc7b4b2ad21201',
+            'duration': 1864.0,
             'uploader': '卡尔',
             'uploader_id': '367138632',
             'upload_date': '20210811',
             'view_count': int,
             'comment_count': int,
             'like_count': int,
+            'tags': 'count:4',
+            'thumbnail': r're:https?://.+\.jpg',
+            'timestamp': 1628675950,
+        },
+    }, {
+        # Only m3u8 available
+        'url': 'https://www.huya.com/video/play/1063345618.html',
+        'info_dict': {
+            'id': '1063345618',
+            'ext': 'mp4',
+            'title': '峡谷第一中！黑铁上钻石顶级教学对抗elo',
+            'categories': ['英雄联盟'],
+            'comment_count': int,
+            'duration': 21603.0,
+            'like_count': int,
+            'thumbnail': r're:https?://.+\.jpg',
+            'timestamp': 1749668803,
+            'upload_date': '20250611',
+            'uploader': '北枫CC',
+            'uploader_id': '2183525275',
+            'view_count': int,
         },
     }]
 
     def _real_extract(self, url: str):
         video_id = self._match_id(url)
-        video_data = self._download_json(
-            'https://liveapi.huya.com/moment/getMomentContent', video_id,
-            query={'videoId': video_id})['data']['moment']['videoInfo']
+        moment = self._download_json(
+            'https://liveapi.huya.com/moment/getMomentContent',
+            video_id, query={'videoId': video_id})['data']['moment']
 
         formats = []
-        for definition in traverse_obj(video_data, ('definitions', lambda _, v: url_or_none(v['url']))):
-            formats.append({
-                'url': definition['url'],
-                **traverse_obj(definition, {
+        for definition in traverse_obj(moment, (
+            'videoInfo', 'definitions', lambda _, v: url_or_none(v['m3u8']),
+        )):
+            fmts = self._extract_m3u8_formats(definition['m3u8'], video_id, 'mp4', fatal=False)
+            for fmt in fmts:
+                fmt.update(**traverse_obj(definition, {
+                    'filesize': ('size', {int_or_none}),
                     'format_id': ('defName', {str}),
-                    'width': ('width', {int_or_none}),
                     'height': ('height', {int_or_none}),
-                    'filesize': ('size', {int_or_none}),
-                }),
-            })
+                    'quality': ('definition', {int_or_none}),
+                    'width': ('width', {int_or_none}),
+                }))
+            formats.extend(fmts)
+        if not formats:
+            self.raise_no_formats('No video found')
```

Comment:
no point to this if it's not expected; core code will `raise_no_formats` if we return an empty formats list
```suggestion
```